libparsec/crates/platform_mountpoint/src/unix/mod.rs

```diff
@@ -0,0 +1,436 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use fuser::{
+    FileAttr, FileType, Filesystem, MountOption, ReplyAttr, ReplyCreate, ReplyData, ReplyDirectory,
+    ReplyEmpty, ReplyEntry, ReplyOpen, ReplyWrite, Request, Session, SessionUnmounter,
+};
+use libc::{ENAMETOOLONG, ENOENT};
+use libparsec_types::{anyhow, FileDescriptor};
+use std::{
+    ffi::OsStr,
+    path::{Path, PathBuf},
+    time::Duration,
+};
+
+use crate::{
+    error::MountpointError, EntryInfo, EntryInfoType, MountpointInterface, MountpointManager,
+    WriteMode,
+};
+
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct Counter(pub u64);
+
+impl Counter {
+    pub fn increment(&mut self) {
+        self.0 += 1;
+    }
+
+    pub fn decrement(&mut self, nlookup: u64) {
+        self.0 -= nlookup;
+    }
+
+    pub fn is_zero(&mut self) -> bool {
+        self.0 == 0
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct Inode(pub u64);
```

Comment:
Add doc comments to explain why this inode exsists (i.e. required by the fuse high-level API and used by fuser functions implemented below)

Reply:
Done